content/projects/gsoc/2020/projects/git-plugin-performance.adoc

```diff
@@ -23,44 +24,94 @@ links:
 
-image:/images/post-images/gsoc-git-performance-improvement/git-additional-behavior.png[title="Performance improvement as an opt-in feature" role="center" width=1000,height=600]
+The graphical results for various benchmarks performed throughout the projects are present in: +
 
-=== Fixing known issues to improve performance link:https://issues.jenkins-ci.org/browse/JENKINS-49757[JENKINS-49757]
-On an agent, the git plugin checks out copies of remote git repositories and it does so by creating empty git repositories (git init) in the workspace, configures them with “git config” and fills them with “git fetch”. After adding a new behavior of honoring refspec on initial clone, the git-plugin faced an issue of a redundant fetch after the initial git-clone operation (which is actually a git init + git fetch) operation. This section of code is a hot code path.
+  * link:/blog/2020/07/2020-07-09-git-performance-improvement-phase1.adoc[Git fetch benchmark analysis]
+  * link:/blog/2020/07/2020-07-29-git-performance-improvement-phase2.adoc[Benchmarks in multiple environments]
 
-link:https://github.com/jenkinsci/git-plugin/pull/845[PR-845] describes an attempt to fix this issue.
+=== Performance enhancement in the checkout step
+The project has introduced a new feature called the `GitToolChooser` which recommends the optimal git implementation for a particular scenario. The user doesn't have to worry about choosing the correct implementation for the given remote repository +
+Here is a graphical representation of the feature:
 
-=== Benchmarking strategy and preliminary results
+image:/images/post-images/gsoc-git-performance-improvement/git-perf-improv.png[title="Performance improvements in the git plugin" role="center" width=1000,height=600]
 
-Here is the https://github.com/jenkinsci/git-client-plugin/pull/521 which provides the code and results from the initial investigation of performance comparison between CliGitAPIImpl and JGitAPIImpl.
+=== Evaluation Phase 1
+
+. Integrate JMH benchmarks module into the git client plugin +
+  - Choose a git operation and write a benchmark
+  - Validate the benchmark results
+  - Repository: https://github.com/jenkinsci/git-client-plugin/tree/master/src/test/java/jmh/benchmark
+
+. Fix git redundant fetch issue link:https://issues.jenkins-ci.org/browse/JENKINS-49757[JENKINS-49757]
+  - Create a fix and write automated test cases
+  - Interactive testing before pushing it to production
+
+. Design a strategy to implement performance improvements in the plugin
+
+==== Resources
+
+. Blog: link:/blog/2020/07/2020-07-09-git-performance-improvement-phase1.adoc[Coding phase 1 summary]
+. Slides: https://docs.google.com/presentation/d/1kaSuYWLBnSvWJWq4lkH5ecOdN-m8n-uymdYHVXcv-EU/edit?usp=sharing
+. Demo: https://youtu.be/HQLhakpx5mk?t=1791
+
+=== Evaluation Phase 2
```

Comment:
Reference pull requests?

Reply:
I agree with Oleg.  For the benefit of others reading (including the Google Summer of Code team), links to the pull requests should be included in the document.

Reply:
Makes total sense, I have added them.